packages/instant/src/containers/selected_asset_button.tsx

```diff
@@ -0,0 +1,36 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { SecondaryButton } from '../components/secondary_button';
+import { State } from '../redux/reducer';
+import { AsyncProcessState } from '../types';
+
+import { SelectedAssetBuyButton } from './selected_asset_buy_button';
+import { SelectedAssetRetryButton } from './selected_asset_retry_button';
+
+interface ConnectedState {
+    buyOrderState: AsyncProcessState;
+}
+export interface SelectedAssetButtonProps {}
+const mapStateToProps = (state: State, _ownProps: SelectedAssetButtonProps): ConnectedState => ({
+    buyOrderState: state.buyOrderState,
+});
+
+const SelectedAssetButtonPresentationComponent: React.StatelessComponent<{
+    buyOrderState: AsyncProcessState;
+}> = props => {
+    if (props.buyOrderState === AsyncProcessState.FAILURE) {
+        return <SelectedAssetRetryButton />;
+    } else if (props.buyOrderState === AsyncProcessState.SUCCESS) {
+        return <SecondaryButton text="Success" isDisabled={true} />;
```

Comment:
Will make this and processing buttons nicer in separate PR

Reply:
This is great -- I would just expect the presentational component to be in `components` and just have the wiring in `containers`.

Also `BuyOrderStateButton` and `SelectedAssetBuyOrderStateButton`?

Reply:
Thanks!  Putting this into a separate file feels like overkill to me: it's a very simple component and will not be reused anywhere else.  Having this defined inline like it is now feels easier to understand, and reduces having to jump into another file.  If this component were to grow larger, be more complex, or be more reusable, I'd be more up for putting it into a separate file.  I'm open to discuss more if you feel strongly though.

Also the `SelectedAsset[component name]` pattern is feeling a bit verbose to me.  It feels more natural to me to follow a pattern like `Connected[component name]`: wondering if you think that is too vague though?

Reply:
Cool, will consider a name and update if I can think of something better than `Connected...`. 

I will move the component to it's own file to follow the convention

Reply:
For the naming, what do you think about this and buttons like the retry button having the prefix of `BuyOrder..` or `CurrentBuyOrder` ?  I feel like these buttons are more closely related to the buy order than they are the selected asset